app/services/matchers/party_name_splitter.rb

```diff
@@ -3,10 +3,9 @@ module Matchers
   class PartyNameSplitter
     attr_reader :party, :full_name, :split_name, :split_count
 
-    def initialize(party, full_name)
+    def initialize(party)
       @party = party
-      @full_name = full_name
-      @split_name = full_name.split(/\p{Space}\p{Space}/)
+      @split_name = party.full_name.squish.split(' ')
```

Comment:
looks like regex which isn't fun, splitting double spaces like how you did looks way better